index.js

```diff
@@ -1,293 +1,324 @@
-tickers['LocalbitcoinsARS'] = null;
-tickers['CoinbaseARS'] = null;
-tickers['BitexARS'] = null;
-tickers['LocalbitcoinsCLP'] = null;
-tickers['BitsoMXN'] = null;
-tickers['BitXZAR'] = null;
-tickers['CoinbaseZAR'] = null;
-tickers['Bit2CILS'] = null;
-tickers['BTCeLitecoin'] = null;
-tickers['KrakenLitecoin'] = null;
-tickers['VircurexLitecoin'] = null;
-tickers['BTCeLitecoinUSD'] = null;
-tickers['BTCeLitecoinEUR'] = null;
-tickers['VircurexWorldcoin'] = null;
-tickers['CryptsyDogecoin'] = null;
-tickers['KrakenDogecoin'] = null;
-tickers['BTCeNamecoinUSD'] = null;
-tickers['CryptsyAuroracoin'] = null;
-tickers['CryptsyBlackcoin'] = null;
-tickers['CryptsyNxt'] = null;
-tickers['PoloniexNxt'] = null;
-tickers['PoloniexBitshares'] = null;
-tickers['KrakenRipple'] = null;
-tickers['PoloniexMaidsafe'] = null;
-tickers['PoloniexBitcoindark'] = null;
-tickers['PoloniexMonero'] = null;
-tickers['CryptsyDashBTC'] = null;
-tickers['CryptsyDashUSD'] = null;
-tickers['BitFinexDashBTC'] = null;
-tickers['BitFinexDashUSD'] = null;
-tickers['PoloniexBurst'] = null;
-const MAX_TICKERS = Object.keys(tickers).length;
 const DEBUG = false
-var ticker_creators = new Array(); // Store all tickers creators here
-var ordered_tickers = new Array();
-var last_ticker_position = 0;
+const DATA_PROVIDERS_URL = "https://raw.githubusercontent.com/neoranga55/bitcoin-price-ticker/refactor-to-use-firefox-frames/data/data-providers.json"
+const ADDON_UPDATE_DOCUMENT_URL = "http://neoranga55.github.io/bitcoin-price-ticker/"
+
+// The main module of the Add-on.
+var ui = require('sdk/ui')
+const {Cc, Ci, Cu} = require("chrome")
+Cu.import("resource://gre/modules/AddonManager.jsm") // Addon Manager required to know addon version
+const setTimeout = require("sdk/timers").setTimeout
+const setInterval = require("sdk/timers").setInterval
+const clearInterval = require("sdk/timers").clearInterval
+const ADDON_ID = "jid0-ziK34XHkBWB9ezxd4l9Q1yC7RP0@jetpack"
+const DEFAULT_REFRESH_RATE = 60
+const DEFAULT_FONT_SIZE = 14
+
+var Preferences = require('sdk/simple-prefs')
+var prefs = Cc["@mozilla.org/preferences-service;1"].getService(Ci.nsIPrefService).getBranch("extensions.ADDON_ID.")
+var Request = require("sdk/request").Request
+var tabs = require("sdk/tabs")
+
+var orderedTickers = new Array()
+var tickers = {} // Store all tickers here
 
 exports.main = function() {
 
-  var getBackgroundColor = function(id) {
-    var low_id = id.toLowerCase();
-    var other_bg_cryptos = [ 'dogecoin', 'worldcoin', 'namecoin', 'auroracoin', 'blackcoin', 'nxt',
-      'bitshares', 'ripple', 'maidsafe', 'bitcoindark', 'monero', 'dash', 'burst' ];
-    for (var i in other_bg_cryptos) {
-      if (low_id.indexOf(other_bg_cryptos[i]) != -1) {  // Alt-coin
+  function getPreference(prefName, type) {
+    if (typeof Preferences.prefs[prefName] == "undefined") {
+      if (DEBUG) console.log("bitcoin-price-ticker addon error: " + prefName + " preference is not defined")
```

Comment:
Line is too long.<br>Expected '{' and instead saw 'console'.<br>Missing semicolon.